### PR TITLE
Improve documentation with FAQ

### PR DESCRIPTION
**Pull request by hardikdr**
**What this PR does / why we need it**: Improve documentation with FAQ.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/machine-controller-manager/issues/121 #529 

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Improved documentation with FAQ.md
```

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+* You must also check the logs of the MCM pod to understand if any logical flow of reconciliation is broken.
+
+
+# Developer
+
+### How should I test my code before submitting a PR?
+
+You can locally setup the MCM using following [guide](https://github.com/gardener/machine-controller-manager/blob/master/docs/development/local_setup.md). You must also enhance the unit-tests related to your changes. You can locally run the unit-test by executing:
+```
+ginkgo --cover pkg/controller/.
```
Comment (prashanth26) [inline docs/FAQ.md:205]:
```suggestion
make test-unit
```

File: README.md [hunk 2]
```diff
@@ -4,7 +4,21 @@
 [![CI Build status](https://concourse.ci.gardener.cloud/api/v1/teams/gardener/pipelines/machine-controller-manager-master/jobs/master-head-update-job/badge)](https://concourse.ci.gardener.cloud/teams/gardener/pipelines/machine-controller-manager-master/jobs/master-head-update-job)
 [![Go Report Card](https://goreportcard.com/badge/github.com/gardener/machine-controller-manager)](https://goreportcard.com/report/github.com/gardener/machine-controller-manager)
 
-Machine Controller Manager (MCM) manages VMs as another kubernetes custom resource. It provides a declarative way to manage VMs. The current implementation supports AWS, GCP, Azure, Alicloud, Packet and Openstack. It can easily be extended to support other cloud providers as well.
+Machine Controller Manager (MCM) manages VMs as another kubernetes custom resource. It provides a declarative way to manage VMs. 
+
+MCM supports following providers:
```
Comment (hardikdr) [inline README.md:9]: I'd like to add a link to the respective providers here, external for OOT internal for in-tree, wdyt? @prashanth26

**Comment by hardikdr** [comment]
@AxiomSamarth have ideas/suggestions to improve, would prefer to wait for him to take a look.

File: README.md (hunk 2 above)
Comment (prashanth26) [reply README.md:9]: Sure makes sense.

**Comment by AxiomSamarth** [comment]
> @AxiomSamarth have ideas/suggestions to improve, would prefer to wait for him to take a look.

@hardikdr @prashanth26 Hi guys, thanks for the opportunity to review the doc. Please find my changes in my PR to the docs branch of Hardik's fork here at https://github.com/hardikdr/machine-controller-manager/pull/5

File: README.md (hunk 2 above)
Comment (hardikdr) [reply README.md:9]: I later felt, let's add the links later when we have OOT drivers available, else most of them will point to in-tree at the moment.

**Review by AxiomSamarth — APPROVED** [review]

**Review by prashanth26 — APPROVED** [review]
/lgtm

**Pull request merged**